app-creator/handlers/templates/templates_test.go

```diff
@@ -0,0 +1,42 @@
+package handlers
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"log"
+	"cloud.google.com/go/datastore"
+	"os"
+	"io/ioutil"
+)
+
+func TestGetTemplates(t *testing.T) {
+	req, err := http.NewRequest("GET", "localhost:8080/api/v1/templates", nil)
+	if err != nil {
+		t.Fatalf("could not create request: %v", err)
+	}
+
+	rec := httptest.NewRecorder()
+
+	l := log.New(os.Stdout, "app-creator", log.LstdFlags)
+
+	db, err := datastore.NewClient(req.Context(), os.Getenv("GOOGLE_CLOUD_PROJECT"))
```

Comment:
You don't want to rely on the actual datastore client in the test, instead, you should use a stub.
Create an interface for the `db` client that includes the `GetAll` method, then inject a custom object that implements `GetAll` returning fake JSON templates. 

This way, you can also test whether the system properly handles errors by adding a test where `GetAll` throws an error instead, and check that the http server returns a 500 error.